src/AppBundle/Controller/UserController.php

```diff
@@ -73,4 +74,21 @@ public function avatarAction($username, Request $request)
 
         return $response;
     }
+
+    /**
+     * @Route("/register/resend-email", name="register_resend_email", methods={"POST"})
+     * @Template()
+     */
+    public function resendRegistrationEmailAction(Request $request, UserManagerInterface $userManager, MailerInterface $mailer)
+    {
+        if($request->request->get("email") !== NULL) {
+            $email = $request->request->get("email");
+            $user = $userManager->findUserByEmail($email);
+            $mailer->sendConfirmationEmailMessage($user);
+        }
+
+        return $this->redirectToRoute('fos_user_registration_check_email');
```

Comment:
It works because you have modified locally `RegistrationController` 🙂 

https://github.com/FriendsOfSymfony/FOSUserBundle/blob/5d2145777fa1d61ba27ba4c1e7016f3c3e7ee7ba/Controller/RegistrationController.php#L106-L124

Before redirecting, you need to set something in session, like in `EmailConfirmationListener`

https://github.com/FriendsOfSymfony/FOSUserBundle/blob/5d2145777fa1d61ba27ba4c1e7016f3c3e7ee7ba/EventListener/EmailConfirmationListener.php#L59-L75